.github/workflows/release.yaml

```diff
@@ -8,21 +8,33 @@ on:
       - .github/renovate.json5
       - pyproject.toml
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
+  get-version:
+    uses: ./.github/workflows/get-version.yaml
+
   ci:
     name: Build & Test
     uses: ./.github/workflows/build-and-test.yaml
     secrets: inherit
+    needs:
+      - get-version
+    with:
+      snap_channel: ${{ needs.get-version.outputs.version }}/edge
```

Comment:
To me. this should be named `snap_risk_level` instead of `snap_channel`, as the snap channel should always align with what is recorded in the version file while here we would only want to control for managing the risk level of the snap to be installed.

Reply:
the `version` file represents the version of the charm and not necessarily the snap. However, to couple them, is the reason I had only put in the config param for `risk_level` and not version, to be able to change only the risk level and not the snap version being installed. Also in our other repos, the `.base_version` only specifies the version and not the risk level anyway.

Reply:
+1, this aligns with the rationality I mentioned earlier. Currently, the charm installs the snap using the following:
```
self.ams.install(channel=f"{CHARM_VERSION}/{snap_risk_level}", revision=revision)
```

The CHARM_VERSION is obtained from the version file, which is handled by the charm itself. Therefore, I fail to see the necessity of reading the version file in the `get-version` workflow. This leads me to believe that we only require the `snap_risk_level` here. If we agree on this, when passing the parameter to the `build-and-test` workflow, the following `cut` command is unnecessary:
```
if [ -n "${{ inputs.snap_channel }}" ]; then
    risk_level="$(echo ${{ inputs.snap_channel }} | cut -d'/' -f2)"
    args="${args} --snap-risk-level ${risk_level}"
fi
```

That would make the workflow simpler.